### ISSUE TITLE
add params kwarg to scrapy.Request()

### ISSUE THREAD

**Issue by pawelmhm** [scrapy/scrapy#2607]
python-requests have useful params kwargs, you can do

```python
params = {
    'expand': 'variations,informationBlocks,customisations',
}

response = requests.request(
    method='GET',
    url=url,
    headers=headers,
    params=params,
)
``

what do you think about adding params kwarg to scrapy,Request()? It would simplify work, there would be no need to urlencode querystring and concatenate strings for url.

**Comment by ramwin** [comment]
Yes, I'm looking for this attribute either. I used to use params in `requests` lib, sometime, it is inconvenient and terrible to url encode the `<spance>, &, ?, %` yourself.

**Comment by grammy-jiang** [comment]
Hi, @pawelmhm 

@leonardfrank provides a way to do this, and if you want to know more, please read [here](https://doc.scrapy.org/en/latest/topics/request-response.html#formrequest-objects).

I would like to provide another solution for this, the module coming with python: [21.8. urllib.parse — Parse URLs into components — Python 3.6.5 documentation](https://docs.python.org/3/library/urllib.parse.html). In this module there are several functions related to URL:

- urlparse
- urlunparse
- parse_qs
- parse_qsl
- urljoin

and many others. Going through this official document will help you a lot.
